Formula/yh.rb

```diff
@@ -0,0 +1,25 @@
+class Yh < Formula
+  desc "YAML syntax highlighter to bring colours where only jq could"
+  homepage "https://github.com/andreazorzetto/yh"
+  url "https://github.com/andreazorzetto/yh/archive/v0.4.0.tar.gz"
+  sha256 "78ef799c500c00164ea05aacafc5c34dccc565e364285f05636c920c2c356d73"
+  license "Apache-2.0"
+
+  head "https://github.com/andreazorzetto/yh.git"
+
+  livecheck do
+    url "https://github.com/andreazorzetto/yh/releases/latest"
+    regex(%r{href=.*?/tag/v((0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*))?}i)
+  end
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", "-o", "yh"
+    bin.install "yh"
+  end
+
+  test do
+    system "#{bin}/yh", "version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Thanks for the guidance. It should be ok now.